manifests/install.pp

```diff
@@ -155,13 +195,81 @@
   }
   # If user did not specify $gitlab_download_link
   else {
+    info('No download link specified')
+  }
+  #Check if v7.10.0 or greater (major packaging changes)
+  if $::gitlab::gitlab_branch >= '7.10.0' {
+    case $::gitlab::gitlab_release {
+      undef, 'basic' : {
+        #Install via packagecloud repository
+        #wget required for installed gpg key in packagecloud repo (may want to use stdlib ensure_packages instead
+        package { 'wget':
+          ensure => 'installed',
+        }
+        include packagecloud
+        #Check OS
+        case $::osfamily {
+          'Debian': {
+            packagecloud::repo { "gitlab/gitlab-ce":
```

Comment:
This is now gitlab's preferred installation method for puppet installs according to the documentation found here:  https://packages.gitlab.com/gitlab/gitlab-ce/install.